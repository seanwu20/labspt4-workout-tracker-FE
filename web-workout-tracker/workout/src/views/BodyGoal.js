import React from "react"

import OnboardingHeader from "../components/OnboardingHeader";
import ButtonWithBackground from "../components/ButtonWithBackground";


class BodyGoal extends React.Component {
    state= {
        goal: ""
    }

    render() {
        // change to real data, when design will be finalized
        const dummyData = [
            {
                id: 1,
                url: "https://images.pexels.com/photos/949130/pexels-photo-949130.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260",
                text: "Slim down"
            },
            {
                id: 2,
                url: "https://images.pexels.com/photos/949130/pexels-photo-949130.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260",
                text: "Get big"
            },
            {
                id: 3,
                url: "https://images.pexels.com/photos/949130/pexels-photo-949130.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260",
                text: "Maintain"
            },
            {
                id: 4,
                url: "https://images.pexels.com/photos/949130/pexels-photo-949130.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260",
                text: "Athletic"
            }
        ]
        return (
            <>
                {/* Reusable header component (Icon + text). Props: icon source(=url) and text(=text) */}

                <OnboardingHeader url="https://img.icons8.com/carbon-copy/100/000000/uefa-euro-trophy.png" text="What's your goal?" />

                {/* Reusable button with background component (background image + text). Needs props for text and image source, onClick change filter, save goal to state */}
                
                {dummyData.map(elem =>
                    <ButtonWithBackground key={elem.id} url={elem.url} text={elem.text} />
                )}

                {/* "Select" Button */}

                {/* @TO-DO: Add possibility to skip onboarding when it grows in next versions of the app */}

            </>
        )
    }
}

export default BodyGoal;