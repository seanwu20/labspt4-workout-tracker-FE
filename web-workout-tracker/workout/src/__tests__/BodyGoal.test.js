import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

import { BodyGoal } from '../views/BodyGoal.js';

it('should render', () => {
    render(<BodyGoal />);
});

it('renders a header "What\'s your goal?"', () => {
    const { queryByTestId } = render(<BodyGoal />);
    const renderedHeader = queryByTestId("body-goal-header").textContent;
    const requiredText = "What's your goal?";
    expect(renderedHeader).toEqual(requiredText);
});

it('renders 8 body goals - 4 for each view', () => {
    const { queryAllByTestId } = render(<BodyGoal />);
    const goalsButtons = queryAllByTestId("body-goal");
    expect(goalsButtons.length).toBe(8); // 4 for desktop and 4 for mobile view
});

it('renders 1 option button', () => {
    const { queryAllByTestId } = render(<BodyGoal />);
    const optionsButtons = queryAllByTestId("option-button");
    expect(optionsButtons.length).toBe(1); // "Select" button is not rendering from the start 
});

it('renders 2 option buttons when user choose body goal option', () => {
    const { queryAllByTestId } = render(<BodyGoal />);
    const goalButton = queryAllByTestId("body-goal")[0];
    fireEvent.click(goalButton);

    const optionsButtons = queryAllByTestId("option-button");
    expect(optionsButtons.length).toBe(2);
});