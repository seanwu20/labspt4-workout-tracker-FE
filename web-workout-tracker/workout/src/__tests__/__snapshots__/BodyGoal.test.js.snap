// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-EHOje fsnBf"
>
  <div
    className="sc-bdVaJa dkPazZ"
  >
    <img
      alt="What's your goal?"
      src="Trophy.svg"
      style={
        Object {
          "marginRight": "7px",
          "width": "20px",
        }
      }
    />
    <h1
      className="sc-bwzfXH iHOuhs"
      data-testid="body-goal-header"
    >
      What's your goal?
    </h1>
  </div>
  <div
    className="sc-bZQynM iZmoKX"
  >
    <button
      className="sc-htpNat eISZRH"
      data-testid="body-goal"
      onBlur={[Function]}
      onClick={[Function]}
      onTouchStart={[Function]}
      opacity="0.7"
      type="desktop"
    >
      <img
        alt=""
        src="CheckMark.svg"
        style={
          Object {
            "height": "18px",
            "left": "12px",
            "position": "absolute",
            "top": "12px",
            "width": "19px",
          }
        }
      />
      <div
        className="sc-bxivhb jrRmzq"
      >
        Slim down
      </div>
    </button>
    <button
      className="sc-htpNat cdigah"
      data-testid="body-goal"
      onBlur={[Function]}
      onClick={[Function]}
      onTouchStart={[Function]}
      opacity="0.7"
      type="mobile"
    >
      <img
        alt=""
        src="CheckMark.svg"
        style={
          Object {
            "height": "18px",
            "left": "12px",
            "position": "absolute",
            "top": "12px",
            "width": "19px",
          }
        }
      />
      <div
        className="sc-bxivhb jrRmzq"
      >
        Slim down
      </div>
    </button>
    <button
      className="sc-htpNat hneIYw"
      data-testid="body-goal"
      onBlur={[Function]}
      onClick={[Function]}
      onTouchStart={[Function]}
      opacity="0.7"
      type="desktop"
    >
      <img
        alt=""
        src="CheckMark.svg"
        style={
          Object {
            "height": "18px",
            "left": "12px",
            "position": "absolute",
            "top": "12px",
            "width": "19px",
          }
        }
      />
      <div
        className="sc-bxivhb jrRmzq"
      >
        Get big
      </div>
    </button>
    <button
      className="sc-htpNat gCToSQ"
      data-testid="body-goal"
      onBlur={[Function]}
      onClick={[Function]}
      onTouchStart={[Function]}
      opacity="0.7"
      type="mobile"
    >
      <img
        alt=""
        src="CheckMark.svg"
        style={
          Object {
            "height": "18px",
            "left": "12px",
            "position": "absolute",
            "top": "12px",
            "width": "19px",
          }
        }
      />
      <div
        className="sc-bxivhb jrRmzq"
      >
        Get big
      </div>
    </button>
    <button
      className="sc-htpNat iKvSkk"
      data-testid="body-goal"
      onBlur={[Function]}
      onClick={[Function]}
      onTouchStart={[Function]}
      opacity="0.7"
      type="desktop"
    >
      <img
        alt=""
        src="CheckMark.svg"
        style={
          Object {
            "height": "18px",
            "left": "12px",
            "position": "absolute",
            "top": "12px",
            "width": "19px",
          }
        }
      />
      <div
        className="sc-bxivhb jrRmzq"
      >
        Maintain
      </div>
    </button>
    <button
      className="sc-htpNat jSRHaN"
      data-testid="body-goal"
      onBlur={[Function]}
      onClick={[Function]}
      onTouchStart={[Function]}
      opacity="0.7"
      type="mobile"
    >
      <img
        alt=""
        src="CheckMark.svg"
        style={
          Object {
            "height": "18px",
            "left": "12px",
            "position": "absolute",
            "top": "12px",
            "width": "19px",
          }
        }
      />
      <div
        className="sc-bxivhb jrRmzq"
      >
        Maintain
      </div>
    </button>
    <button
      className="sc-htpNat jbMiBB"
      data-testid="body-goal"
      onBlur={[Function]}
      onClick={[Function]}
      onTouchStart={[Function]}
      opacity="0.7"
      type="desktop"
    >
      <img
        alt=""
        src="CheckMark.svg"
        style={
          Object {
            "height": "18px",
            "left": "12px",
            "position": "absolute",
            "top": "12px",
            "width": "19px",
          }
        }
      />
      <div
        className="sc-bxivhb jrRmzq"
      >
        Athletic
      </div>
    </button>
    <button
      className="sc-htpNat khPDjk"
      data-testid="body-goal"
      onBlur={[Function]}
      onClick={[Function]}
      onTouchStart={[Function]}
      opacity="0.7"
      type="mobile"
    >
      <img
        alt=""
        src="CheckMark.svg"
        style={
          Object {
            "height": "18px",
            "left": "12px",
            "position": "absolute",
            "top": "12px",
            "width": "19px",
          }
        }
      />
      <div
        className="sc-bxivhb jrRmzq"
      >
        Athletic
      </div>
    </button>
  </div>
  <div
    className="sc-gzVnrw ihfQly"
  >
    <button
      className="sc-ifAKCX BYpoi"
      data-testid="option-button"
    >
      I'll do this later
    </button>
  </div>
</div>
`;
